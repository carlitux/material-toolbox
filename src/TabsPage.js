// @flow
import * as React from 'react';

import { LayoutCell } from './material-toolbox/layout-grid';
import { Text } from './material-toolbox/typography';
import BasicTabBar from './tabs-examples/BasicTabBar';
import ThemedLink from './ThemedLink';

type Props = {};

export default class TabsPage extends React.Component<Props> {
  componentDidMount() {
    window.scrollTo(0, 0);
  }

  render() {
    return (
      <LayoutCell span={12}>
        <Text component="h1" textStyle="display1">
          Tabs
        </Text>

        <Text component="h2" textStyle="headline">
          Tabs components is a React wrapper of mdc-tabs component.
        </Text>

        <Text component="h3" textStyle="title">
          Design & API Documentation
        </Text>

        <p>
          <ThemedLink
            to="https://material.io/guidelines/components/snackbars-toasts.html"
            target="_blank">
            Material Design guidelines: Snackbars & toasts
          </ThemedLink>
        </p>

        <p>
          <ThemedLink
            to="https://material.io/components/web/catalog/snackbars/"
            target="_blank">
            Material Components Web Css and Scss customization
          </ThemedLink>
        </p>

        <Text component="h1" textStyle="display1">
          React components
        </Text>

        <BasicTabBar />

        {/* <Text component="p" textStyle="body1"> */}
        {/*   Material in motion is responsive and natural. Use these easing curves and */}
        {/*   duration patterns to create smooth and consistent motion. */}
        {/* </Text> */}
        {/* <Text component="p" textStyle="body2"> */}
        {/*   So this is not implemented as React component this should be installed and */}
        {/*   used as you need in your components.{' '} */}
        {/*   <ThemedLink */}
        {/*     target="_blank" */}
        {/*     to="https://material.io/components/web/catalog/animation/"> */}
        {/*     Read This. */}
        {/*   </ThemedLink> */}
        {/* </Text> */}
      </LayoutCell>
    );
  }
}
