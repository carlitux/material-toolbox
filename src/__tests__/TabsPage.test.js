// @flow
import React from 'react';
import { shallow } from 'enzyme';
import TabsPage from '../TabsPage';

describe('page::TabsPage', () => {
  it('Should match snapshoot', () => {
    const scrollTo = jest.fn();
    window.scrollTo = scrollTo;
    const component = shallow(<TabsPage />);
    expect(component).toMatchSnapshot();
    expect(scrollTo.mock.calls).toHaveLength(1);
    delete window.scrollTo;
  });
});
