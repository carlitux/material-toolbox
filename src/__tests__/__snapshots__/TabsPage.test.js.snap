// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page::TabsPage Should match snapshoot 1`] = `
<Cell
  span={12}
>
  <Text
    adjustMargin={false}
    component="h1"
    textStyle="display1"
  >
    Tabs
  </Text>
  <Text
    adjustMargin={false}
    component="h2"
    textStyle="headline"
  >
    Tabs components is a React wrapper of mdc-tabs component.
  </Text>
  <Text
    adjustMargin={false}
    component="h3"
    textStyle="title"
  >
    Design & API Documentation
  </Text>
  <p>
    <ThemedLink
      component="a"
      target="_blank"
      to="https://material.io/guidelines/components/snackbars-toasts.html"
    >
      Material Design guidelines: Snackbars & toasts
    </ThemedLink>
  </p>
  <p>
    <ThemedLink
      component="a"
      target="_blank"
      to="https://material.io/components/web/catalog/snackbars/"
    >
      Material Components Web Css and Scss customization
    </ThemedLink>
  </p>
  <Text
    adjustMargin={false}
    component="h1"
    textStyle="display1"
  >
    React components
  </Text>
  <BasicTabBar />
</Cell>
`;
